### PR TITLE
clean up docs, tests, and refactor read pandas

### PR DESCRIPTION
**Pull request by oscarbatori**
Some changes:
- remove Sphinx and docs
- remove redundant tests that now are in doltcli
- make Pandas reads use new doltcli ability to inject file parser to improve perf
- bump doltcli dep

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -27,7 +27,7 @@ SQLAlchemy = "1.3.18"
 wcwidth = "0.2.5"
 more-itertools = "^8.6.0"
 mysql-connector-python = "^8.0.20"
-doltcli = "0.1.2"
+doltcli = "^0.1.4"
```
Comment (max-hoffman) [inline pyproject.toml:30]: did we skip a version?

**Review by max-hoffman — APPROVED** [review]
LGTM

File: pyproject.toml (hunk 1 above)
Comment (oscarbatori) [reply pyproject.toml:30]: I committed `1.0.3` forgetting that the release process bumped the version... seemed inconsequential to skip a patch increment.

Comment (max-hoffman) [reply pyproject.toml:30]: skipping 0.1.3 is fine, releasing 1.0.3 would be somewhat disruptive

**Pull request merged**